src/controller.cpp

```diff
@@ -0,0 +1,68 @@
+#include "controller.hpp"
+
+MotorController::Controller::Controller() : dcInterface(), stepperInterface() {
+}
+
+MotorController::Controller::INTERFACE MotorController::Controller::getSelectedInterface() const {
+    // Getting while interface is nullptr
+    if (selectedInterface == nullptr) {
+        hwlib::cout << "Error:: Trying to get an nullptr interface (controller.cpp, line: " << __LINE__ << ")" << hwlib::endl;
+        // Better not continue
+        while (1) {
+        }
+    }
+
+    // Convert the interface to enum
+    if (selectedInterface == &dcInterface) {
+        return MotorController::Controller::INTERFACE::DC;
+    }
+
+    return MotorController::Controller::INTERFACE::STEPPER;
+}
+
+void MotorController::Controller::setSelectedInterface(const MotorController::Controller::INTERFACE newInterface) {
+    // Prevent setting the interface a second time
+    if (selectedInterface != nullptr) {
+        return;
+    }
+
+    switch (newInterface) {

```

Comment:
In my opinion, a switch statement is dangerous (forgetting breaks, which results in unpredictable behavior and may cost a lot of time to debug)
Use ifs and elseifs otherwise.